client/state/data-layer/wpcom/email-forwarding/add/index.js

```diff
@@ -85,6 +87,8 @@ export const addEmailForwardSuccess = ( action, response ) => {
 			);
 		}
 
+		onSuccessRedirect();
```

Comment:
Would it be simpler to add a reducer that monitors the success state and use that value to invoke a redirect in the component itself?

I'm wondering if having this here may be causing the issues with redirect not working.